frontend/src/Frontend/ModuleExplorer/File.hs

```diff
@@ -206,11 +207,15 @@ fetchFile m onFileRef = do
 {-          else pure $ Just req -}
 
 -- | Get the `Module`s contained in `PactFile`.
-fileModules :: PactFile -> Map ModuleName (ModuleDef (Term Name))
+fileModules :: PactFile -> Either Text (Map ModuleName (ModuleDef (Term Name)))
 fileModules (Code c) = case Pact.compileExps (Pact.mkTextInfo c) <$> Pact.parseExprs c of
-  Right (Right terms) -> Map.fromList $ mapMaybe getModule terms
-  _                   -> Map.empty
+  Left err -> Left $ "Parsing failed: " <> tshow err
+  Right (Left err) -> Left $ "Compilation failed: " <> tshow err
+  Right (Right terms) -> Right $ Map.fromList $ mapMaybe getModule terms
 
+-- TODO: remove, only used for compatibility with older code
```

Comment:
Make a ticket for this so we don't forget?

Reply:
Done